### PR TITLE
Add value selection(like for month) for custom fields

### LINKED ISSUE #11109: Add a value selection (like for month) for content selectors in custom entry types
**Is your suggestion for improvement related to a problem? Please describe.**
When I create content selectors for a field, I would like to choose them from a list of values and not having to remember the value.

**Describe the solution you'd like**
A value selection like for the month field: 
grafik

**Additional context**
1. Create a new library
2. Go to Preferences -> Entry types -> Add a new custom field to an entry type e.g. Misc, for example, name it "mycustom"
3. Save and restart Jabref
4. Right-click on the library-> Library properties -> Content selectors ->  Add as in the screenshot below
grafik
5. Add an new entry "Misc"  into the library
6. See the new custom field
7. you have to type a letter e.g. "k" to get the value suggested

### PR DESCRIPTION
**Pull request by azatyamanaev**
Closes #11109 

Added editor class for custom fields. This allows to select values from specified content selectors for corresponding field in entry editor tab. 

When you add a custom field and set content selectors for it, in entry editor tab you can choose from a list of specified values.

1. Add a custom field
1_add_custom_field

2. Set content selectors
2_add_content_selectors

3. Check value selection in entry editor tab

3_check_value_selection

When the added custom field is multiline, standard text area field is used.

1. Change added custom field to multiline
4_change_field_to_multiline

2. Check how it is displayed in entry editor tab
5_check_selection_for_multiline

### Mandatory checks

- [x] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [x] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [x] Screenshots added in PR description (for UI changes)
- [x] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [x] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,6 +11,9 @@ Note that this project **does not** adhere to [Semantic Versioning](https://semv
 
 ### Added
 
+- We added a new keyboard shortcut  <kbd>ctrl</kbd> + <kbd>,</kbd> to open the preferences. [#11154](https://github.com/JabRef/jabref/pull/11154)
+- [azatayamanaev](https://github.com/azatyamanaev) added value selection(like for month) for content selectors in custom entry types. [#11109](https://github.com/JabRef/jabref/issues/11109)
```
Comment (Siedlerchr) [inline CHANGELOG.md:15]: We do not include our names here. Names are already part of the commit and committers will be highlighted at the release blog posts

File: src/main/java/org/jabref/gui/fieldeditors/FieldEditors.java [hunk 2]
```diff
@@ -112,7 +114,8 @@ private static SuggestionProvider<?> getSuggestionProvider(Field field, Suggesti
             // Enrich auto completion by content selector values
             try {
                 return new ContentSelectorSuggestionProvider((SuggestionProvider<String>) suggestionProvider, contentSelectorValues);
-            } catch (ClassCastException exception) {
+            } catch (
+                    ClassCastException exception) {
```
[the remark this answers is not in the document]
Comment (koppor) [reply src/main/java/org/jabref/gui/fieldeditors/FieldEditors.java]: @azatyamanaev Do you use IntelliJ 2024.1 or something of 2023?

[the remark this answers is not in the document]
Comment (azatyamanaev) [reply src/main/java/org/jabref/gui/fieldeditors/FieldEditors.java]: Yes, I use 2024.1

**Review by Siedlerchr — APPROVED** [review]
lgtm!

**Pull request merged**